npm install fails on nodejs version 12.5

Hello,

I am having this exception while installing node-gd. The operating system is MacOS Mojave. NodeJS version is 12.5.0 and I have installed pkg-config and gd via homebrew.

Is there anything that I miss? or is the library incompatible with the NodeJS 12?

```
> node-gyp rebuild

  CXX(target) Release/obj.target/node_gd/cpp/addon.o
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:202:
In file included from ../../nan/nan_converters.h:67:
../../nan/nan_converters_43_inl.h:22:1: warning: 'ToBoolean' is deprecated: ToBoolean can never throw. Use
      Local version. [-Wdeprecated-declarations]
X(Boolean)
^
../../nan/nan_converters_43_inl.h:18:12: note: expanded from macro 'X'
      val->To ## TYPE(v8::Isolate::GetCurrent()->GetCurrentContext())          \
           ^
<scratch space>:204:1: note: expanded from here
ToBoolean
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2523:3: note: 'ToBoolean' has been explicitly marked
      deprecated here
  V8_DEPRECATED("ToBoolean can never throw. Use Local version.",
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:311:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:202:
In file included from ../../nan/nan_converters.h:67:
../../nan/nan_converters_43_inl.h:40:1: warning: 'BooleanValue' is deprecated: BooleanValue can never throw.
      Use Isolate version. [-Wdeprecated-declarations]
X(bool, Boolean)
^
../../nan/nan_converters_43_inl.h:37:15: note: expanded from macro 'X'
  return val->NAME ## Value(isolate->GetCurrentContext());                     \
              ^
<scratch space>:211:1: note: expanded from here
BooleanValue
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2561:3: note: 'BooleanValue' has been explicitly marked
      deprecated here
  V8_DEPRECATED("BooleanValue can never throw. Use Isolate version.",
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:311:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:203:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:103:42: error: no viable conversion from 'v8::Isolate *' to
      'Local<v8::Context>'
  return scope.Escape(v8::Function::New( isolate
                                         ^~~~~~~
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:183:7: note: candidate constructor
      (the implicit copy constructor) not viable: no known conversion from 'v8::Isolate *' to
      'const v8::Local<v8::Context> &' for 1st argument
class Local {
      ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:183:7: note: candidate constructor
      (the implicit move constructor) not viable: no known conversion from 'v8::Isolate *' to
      'v8::Local<v8::Context> &&' for 1st argument
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:187:13: note: candidate template ignored: could not
      match 'Local<type-parameter-0-0>' against 'v8::Isolate *'
  V8_INLINE Local(Local<S> that)
            ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:4126:22: note: passing argument to parameter 'context'
      here
      Local<Context> context, FunctionCallback callback,
                     ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:203:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:337:37: error: too few arguments to function call, expected 2, have 1
  return v8::StringObject::New(value).As<v8::StringObject>();
         ~~~~~~~~~~~~~~~~~~~~~      ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:5381:3: note: 'New' declared here
  static Local<Value> New(Isolate* isolate, Local<String> value);
  ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:203:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:337:58: error: expected '(' for function-style cast or type construction
  return v8::StringObject::New(value).As<v8::StringObject>();
                                         ~~~~~~~~~~~~~~~~^
../../nan/nan_implementation_12_inl.h:337:60: error: expected expression
  return v8::StringObject::New(value).As<v8::StringObject>();
                                                           ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:839:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:173:1: note: 'MakeCallback' has been explicitly
      marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:93:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:854:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:166:1: note: 'MakeCallback' has been explicitly
      marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:93:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:869:18: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return node::MakeCallback(
                 ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:159:1: note: 'MakeCallback' has been explicitly
      marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:93:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:916:44: error: no matching member function for call to 'ToString'
      v8::Local<v8::String> string = from->ToString();
                                     ~~~~~~^~~~~~~~
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2528:44: note: candidate function not viable: requires
      single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                           ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2544:31: note: candidate function not viable: requires
      single argument 'isolate', but no arguments were provided
                Local<String> ToString(Isolate* isolate) const);
                              ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:926:37: error: cannot initialize a parameter of type 'v8::Isolate *' with an lvalue of type
      'char *'
        length_ = string->WriteUtf8(str_, static_cast<int>(len), 0, flags);
                                    ^~~~
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2738:26: note: passing argument to parameter 'isolate'
      here
  int WriteUtf8(Isolate* isolate, char* buffer, int length = -1,
                         ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1478:31: warning: 'MakeCallback' is deprecated: Use MakeCallback(..., async_context)
      [-Wdeprecated-declarations]
    return scope.Escape(node::MakeCallback(
                              ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:173:1: note: 'MakeCallback' has been explicitly
      marked deprecated here
NODE_DEPRECATED("Use MakeCallback(..., async_context)",
^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/node.h:93:20: note: expanded from macro 'NODE_DEPRECATED'
    __attribute__((deprecated(message))) declarator
                   ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1538:28: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
    New(persistentHandle)->Set(New(key).ToLocalChecked(), value);
                           ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1544:28: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
    New(persistentHandle)->Set(key, value);
                           ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1550:28: warning: 'Set' is deprecated: Use maybe version [-Wdeprecated-declarations]
    New(persistentHandle)->Set(index, value);
                           ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3367:3: note: 'Set' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1556:32: warning: 'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
        New(persistentHandle)->Get(New(key).ToLocalChecked()));
                               ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3412:3: note: 'Get' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1562:48: warning: 'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
    return scope.Escape(New(persistentHandle)->Get(key));
                                               ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3412:3: note: 'Get' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
../../nan/nan.h:1567:48: warning: 'Get' is deprecated: Use maybe version [-Wdeprecated-declarations]
    return scope.Escape(New(persistentHandle)->Get(index));
                                               ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:3416:3: note: 'Get' has been explicitly marked
      deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(uint32_t index));
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:326:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:2365:
../../nan/nan_object_wrap.h:24:25: error: no member named 'IsNearDeath' in 'Nan::Persistent<v8::Object,
      v8::NonCopyablePersistentTraits<v8::Object> >'
    assert(persistent().IsNearDeath());
           ~~~~~~~~~~~~ ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/assert.h:93:25: note:
      expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:2365:
../../nan/nan_object_wrap.h:67:18: warning: 'MarkIndependent' is deprecated: Weak objects are always
      considered independent. Use TracedGlobal when trying to use EmbedderHeapTracer. Use a strong handle when
      trying to keep an object alive. [-Wdeprecated-declarations]
    persistent().MarkIndependent();
                 ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:563:3: note: 'MarkIndependent' has been explicitly
      marked deprecated here
  V8_DEPRECATED(
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:311:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated(message)))
                            ^
In file included from ../cpp/addon.cc:1:
In file included from ../../nan/nan.h:2365:
../../nan/nan_object_wrap.h:124:26: error: no member named 'IsNearDeath' in 'Nan::Persistent<v8::Object,
      v8::NonCopyablePersistentTraits<v8::Object> >'
    assert(wrap->handle_.IsNearDeath());
           ~~~~~~~~~~~~~ ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/assert.h:93:25: note:
      expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from ../cpp/addon.cc:3:
In file included from ../cpp/node_gd.cc:27:
../cpp/createworker.cc:40:78: error: too few arguments to function call, single argument 'context' was not
      specified
      v8::MaybeLocal<v8::Object> _image = Nan::NewInstance(func->GetFunction(), _argc, _argv);
                                                           ~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:5948:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:291:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(0, width);
  ^~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:292:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(1, height);
  ^~~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:307:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(0, width);
  ^~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:308:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(1, height);
  ^~~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:315:3: error: too few arguments to function call, single argument 'context' was not
      specified
  RETURN_IMAGE(im);
  ^~~~~~~~~~~~~~~~
../cpp/node_gd.cc:144:70: note: expanded from macro 'RETURN_IMAGE'
    MaybeLocal<Object> instance = Nan::NewInstance(func->GetFunction(), argc, argv);\
                                                   ~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:5948:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:320:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(0, width);
  ^~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:321:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(1, height);
  ^~~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:336:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(0, width);
  ^~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:337:3: error: too few arguments to function call, single argument 'context' was not
      specified
  REQ_INT_ARG(1, height);
  ^~~~~~~~~~~~~~~~~~~~~~
../cpp/node_gd.cc:84:29: note: expanded from macro 'REQ_INT_ARG'
  VAR = info[I]->Int32Value();
        ~~~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../cpp/addon.cc:3:
../cpp/node_gd.cc:344:3: error: too few arguments to function call, single argument 'context' was not
      specified
  RETURN_IMAGE(im);
  ^~~~~~~~~~~~~~~~
../cpp/node_gd.cc:144:70: note: expanded from macro 'RETURN_IMAGE'
    MaybeLocal<Object> instance = Nan::NewInstance(func->GetFunction(), argc, argv);\
                                                   ~~~~~~~~~~~~~~~~~ ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8.h:5948:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/buraktamturk/.node-gyp/12.5.0/include/node/v8config.h:351:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
13 warnings and 20 errors generated.
make: *** [Release/obj.target/node_gd/cpp/addon.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/usr/local/Cellar/node/12.5.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! node -v v12.5.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-gd@1.5.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-gd@1.5.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/buraktamturk/.npm/_logs/2019-07-04T18_26_29_996Z-debug.log
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,7 @@ node_js:
   - "8"
   - "10"
   - "11"
+  - "12"
 
 sudo: required
 

--- a/cpp/createworker.cc
+++ b/cpp/createworker.cc
@@ -37,7 +37,7 @@ class CreateWorker : public Nan::AsyncWorker {
       const int _argc = 1;
       v8::Local<v8::Value> _argv[_argc] = {Nan::New<v8::External>(image)};
       v8::Local<v8::FunctionTemplate> func = Nan::New(Gd::Image::constructor);
-      v8::MaybeLocal<v8::Object> _image = Nan::NewInstance(func->GetFunction(), _argc, _argv);
+      v8::MaybeLocal<v8::Object> _image = Nan::NewInstance(func->GetFunction(Nan::GetCurrentContext()).ToLocalChecked(), _argc, _argv);
 
       const int argc = 2;
       v8::Local<v8::Value> argv[argc] = {

--- a/cpp/node_gd.cc
+++ b/cpp/node_gd.cc
@@ -99,7 +99,7 @@ void nodeGdErrorWrapper(int priority, const char *format, va_list args)
   if (info.Length() <= (I) || !info[I]->IsNumber()) {                   \
     return Nan::ThrowTypeError("Argument " #I " must be a number");     \
   }                                                                     \
-  VAR = info[I]->NumberValue();
+  VAR = info[I]->NumberValue(Nan::GetCurrentContext()).FromJust();
 
 #define REQ_EXT_ARG(I, VAR)                                             \
   if (info.Length() <= (I) || !info[I]->IsExternal()) {                 \
@@ -129,7 +129,7 @@ void nodeGdErrorWrapper(int priority, const char *format, va_list args)
   if (info.Length() <= (I)) {                                           \
     VAR = (DEFAULT);                                                    \
   } else if (info[I]->IsBoolean()) {                                    \
-    VAR = info[I]->BooleanValue();                                      \
+    VAR = info[I]->BooleanValue(Nan::GetCurrentContext()).FromJust();   \
   } else {                                                              \
     return Nan::ThrowTypeError("Optional argument " #I " must be a boolean");    \
   }
@@ -141,7 +141,7 @@ void nodeGdErrorWrapper(int priority, const char *format, va_list args)
     const int argc = 1;                                                 \
     Local<Value> argv[argc] = {Nan::New<External>(IMG)};                \
     Local<FunctionTemplate> func = Nan::New(Image::constructor);        \
-    MaybeLocal<Object> instance = Nan::NewInstance(func->GetFunction(), argc, argv);\
+MaybeLocal<Object> instance = Nan::NewInstance(func->GetFunction(Nan::GetCurrentContext()).ToLocalChecked(), argc, argv);\
     info.GetReturnValue().Set(instance.ToLocalChecked());               \
   }
 
@@ -159,12 +159,12 @@ void nodeGdErrorWrapper(int priority, const char *format, va_list args)
     gdImagePtr im;                                                      \
     Local<Value> obj;                                                   \
     if(Buffer::HasInstance(info[0])) {                                  \
-      obj = info[0]->ToObject();                                        \
+      obj = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();    \
       char *buffer_data = Buffer::Data(obj);                            \
       size_t buffer_length = Buffer::Length(obj);                       \
       im = gdImageCreateFrom##TYPE##Ptr(buffer_length, buffer_data);    \
     } else {                                                            \
-      obj = info[0]->ToString();                                        \
+      obj = info[0]->ToString(Nan::GetCurrentContext()).ToLocalChecked();    \
       ssize_t len = Nan::DecodeBytes(obj, Nan::Encoding(BINARY));       \
       char *buf = new char[len];                                        \
       ssize_t written = Nan::DecodeWrite(buf, len, obj);                \
@@ -410,7 +410,7 @@ NAN_METHOD(Gd::TrueColor) {
   REQ_INT_ARG(1, g);
   REQ_INT_ARG(2, b);
 
-  uint32_t result = Nan::New<Uint32>(gdTrueColor(r, g, b))->Uint32Value();
+  uint32_t result = gdTrueColor(r, g, b);
 
   info.GetReturnValue().Set(result);
 }
@@ -422,7 +422,7 @@ NAN_METHOD(Gd::TrueColorAlpha) {
   REQ_INT_ARG(2, b);
   REQ_INT_ARG(3, a);
 
-  uint32_t result = Nan::New<Uint32>(gdTrueColorAlpha(r, g, b, a))->Uint32Value();
+  uint32_t result = gdTrueColorAlpha(r, g, b, a);
 
   info.GetReturnValue().Set(result);
 }
@@ -617,7 +617,7 @@ void Gd::Image::Init(v8::Local<Object> exports) {
   Nan::SetPrototypeMethod(t, "compare", Compare);
 
   constructor.Reset(t);
-  exports->Set(Nan::New<String>("Image").ToLocalChecked(), t->GetFunction());
+    exports->Set(Nan::New<String>("Image").ToLocalChecked(), t->GetFunction(Nan::GetCurrentContext()).ToLocalChecked());
 }
 
 NAN_METHOD(Gd::Image::New) {
@@ -987,8 +987,8 @@ NAN_METHOD(Gd::Image::Polygon) {
     Local<Value> v = array->Get(Nan::New<Integer>(i));
     if (!v->IsObject()) continue;
 
-    Local<Object> o = v->ToObject();
-    if ( !o->Has(x) || !o->Has(y)) continue;
+    Local<Object> o = v->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
+    if ( !o->Has(Nan::GetCurrentContext(), x).FromJust() || !o->Has(Nan::GetCurrentContext(), y).FromJust()) continue;
 
     points[i].x = o->Get(x)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
     points[i].y = o->Get(y)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
@@ -1023,8 +1023,8 @@ NAN_METHOD(Gd::Image::OpenPolygon) {
     Local<Value> v = array->Get(Nan::New<Integer>(i));
     if (!v->IsObject()) continue;
 
-    Local<Object> o = v->ToObject();
-    if ( !o->Has(x) || !o->Has(y)) continue;
+    Local<Object> o = v->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
+    if ( !o->Has(Nan::GetCurrentContext(), x).FromJust() || !o->Has(Nan::GetCurrentContext(), y).FromJust()) continue;
 
     points[i].x = o->Get(x)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
     points[i].y = o->Get(y)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
@@ -1059,8 +1059,8 @@ NAN_METHOD(Gd::Image::FilledPolygon) {
     Local<Value> v = array->Get(Nan::New<Integer>(i));
     if (!v->IsObject()) continue;
 
-    Local<Object> o = v->ToObject();
-    if ( !o->Has(x) || !o->Has(y)) continue;
+    Local<Object> o = v->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
+    if ( !o->Has(Nan::GetCurrentContext(), x).FromJust() || !o->Has(Nan::GetCurrentContext(), y).FromJust()) continue;
 
     points[i].x = o->Get(x)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
     points[i].y = o->Get(y)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
@@ -1545,14 +1545,14 @@ NAN_METHOD(Gd::Image::StringFTEx) {
   stringExtra.flags = 0;
 
   // linespacing
-  if (stringExtraParameter->Has(linespacing)) {
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), linespacing).FromJust()) {
     stringExtra.flags |= gdFTEX_LINESPACE;
-    stringExtra.linespacing = stringExtraParameter->Get(linespacing)->NumberValue();
+      stringExtra.linespacing = stringExtraParameter->Get(linespacing)->NumberValue(Nan::GetCurrentContext()).FromJust();
   }
 
   // charmap
-  if (stringExtraParameter->Has(charmap)) {
-    Local<String> localCharmap = stringExtraParameter->Get(charmap)->ToString();
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), charmap).FromJust()) {
+    Local<String> localCharmap = stringExtraParameter->Get(charmap)->ToString(Nan::GetCurrentContext()).ToLocalChecked();
     stringExtra.charmap = 0;
 
     Nan::Utf8String chmap(localCharmap);
@@ -1576,25 +1576,25 @@ NAN_METHOD(Gd::Image::StringFTEx) {
   // resolution
   // horizontal dpi
   // The JavaScript value of NaN will be cast to a 0 by v8?
-  if (stringExtraParameter->Has(hdpi)) {
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), hdpi).FromJust()) {
     stringExtra.flags |= gdFTEX_RESOLUTION;
     stringExtra.hdpi = stringExtraParameter->Get(hdpi)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
-    if (!stringExtraParameter->Has(vdpi)) {
+    if (!stringExtraParameter->Has(Nan::GetCurrentContext(), vdpi).FromJust()) {
       stringExtra.vdpi = stringExtra.hdpi;
     }
   }
   // vertical dpi
-  if (stringExtraParameter->Has(vdpi)) {
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), vdpi).FromJust()) {
     stringExtra.flags |= gdFTEX_RESOLUTION;
     stringExtra.vdpi = stringExtraParameter->Get(vdpi)->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value();
-    if (!stringExtraParameter->Has(hdpi)) {
+    if (!stringExtraParameter->Has(Nan::GetCurrentContext(), hdpi).FromJust()) {
       stringExtra.hdpi = stringExtra.vdpi;
     }
   }
 
   // disable kerning
-  if (stringExtraParameter->Has(disable_kerning)) {
-    bool is_disable_kerning = stringExtraParameter->Get(disable_kerning)->BooleanValue();
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), disable_kerning).FromJust()) {
+    bool is_disable_kerning = stringExtraParameter->Get(disable_kerning)->BooleanValue(Nan::GetCurrentContext()).FromJust();
     if (is_disable_kerning) {
       stringExtra.flags |= gdFTEX_DISABLE_KERNING;
     }
@@ -1602,23 +1602,23 @@ NAN_METHOD(Gd::Image::StringFTEx) {
 
   // xshow
   bool is_xshow = false;
-  if (stringExtraParameter->Has(xshow)) {
-    is_xshow = stringExtraParameter->Get(xshow)->BooleanValue();
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), xshow).FromJust()) {
+    is_xshow = stringExtraParameter->Get(xshow)->BooleanValue(Nan::GetCurrentContext()).FromJust();
     if (is_xshow) {
       stringExtra.flags |= gdFTEX_XSHOW;
     }
   }
 
   // fontpathname
-  if (stringExtraParameter->Has(fontpath)) {
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), fontpath).FromJust()) {
     Nan::Utf8String localFontpathname(stringExtraParameter->Get(fontpath)->ToString(Nan::GetCurrentContext()).FromMaybe(v8::Local<v8::String>()));
     stringExtra.flags |= gdFTEX_FONTPATHNAME;
     stringExtra.fontpath = *localFontpathname;
   }
 
   // use_fontconfig
-  if (stringExtraParameter->Has(use_fontconfig)) {
-    bool is_use_fontconfig = stringExtraParameter->Get(use_fontconfig)->BooleanValue();
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), use_fontconfig).FromJust()) {
+    bool is_use_fontconfig = stringExtraParameter->Get(use_fontconfig)->BooleanValue(Nan::GetCurrentContext()).FromJust();
     if (is_use_fontconfig) {
       stringExtra.flags |= gdFTEX_FONTCONFIG;
     }
@@ -1626,8 +1626,8 @@ NAN_METHOD(Gd::Image::StringFTEx) {
 
   // return_fontpathname
   bool is_use_fontpathname = false;
-  if (stringExtraParameter->Has(return_fontpathname)) {
-    is_use_fontpathname = stringExtraParameter->Get(return_fontpathname)->BooleanValue();
+  if (stringExtraParameter->Has(Nan::GetCurrentContext(), return_fontpathname).FromJust()) {
+    is_use_fontpathname = stringExtraParameter->Get(return_fontpathname)->BooleanValue(Nan::GetCurrentContext()).FromJust();
     if (is_use_fontpathname) {
       stringExtra.flags |= gdFTEX_RETURNFONTPATHNAME;
       stringExtra.fontpath = NULL;
@@ -1855,7 +1855,7 @@ NAN_SETTER(Gd::Image::InterlaceSetter) {
   Image *im = ObjectWrap::Unwrap<Image>(info.This());
 
   if (value->IsBoolean()) {
-    bool interlace = value->BooleanValue();
+    bool interlace = value->BooleanValue(Nan::GetCurrentContext()).FromJust();
 
     gdImageInterlace(*im, interlace ? 1 : 0);
   }

